etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+*/
+rules[ '@typescript-eslint/no-require-imports' ] = 'off';
+
+/**
+* Disallows aliases for `this`.
+*
+* @name no-this-alias
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-this-alias]{@link https://typescript-eslint.io/rules/no-this-alias}
+*
+* @example
+* // Bad...
+* const self = this;
+* const that = this;
+*
+* // Good...
+* function example() {
+*   console.log(this);
+* }
+*/
+rules[ '@typescript-eslint/no-this-alias' ] = 'error';
+
+/**
+* Disables the rule against unnecessary boolean literal comparisons.
+*
+* @name no-unnecessary-boolean-literal-compare
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unnecessary-boolean-literal-compare]{@link https://typescript-eslint.io/rules/no-unnecessary-boolean-literal-compare}
+*
+* @example
+* // Okay...
+* const x = true;
+* if (x === true) {
+*   // ...
+* }
+*/
+rules[ '@typescript-eslint/no-unnecessary-boolean-literal-compare' ] = 'off';
+
+/**
+* Disallows unnecessary qualifiers (like namespaces or enum names) when the referenced item is accessible without them.
+*
+* @name no-unnecessary-qualifier
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-qualifier]{@link https://typescript-eslint.io/rules/no-unnecessary-qualifier}
+*
+* @example
+* // Bad...
+* namespace A {
+*   export const B = 1;
+* }
+* const x = A.B;
+*
+* // Good...
+* const B = 1;
+* const x = B;
+*/
+rules[ '@typescript-eslint/no-unnecessary-qualifier' ] = 'error';
+
+/**
+* Disallows unnecessary type arguments in generics where the argument is the default.
+*
+* @name no-unnecessary-type-arguments
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-arguments]{@link https://typescript-eslint.io/rules/no-unnecessary-type-arguments}
+*
+* @example
+* // Bad...
+* const x: Array<number> = [];
+*
+* // Good...
+* const x: number[] = [];
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-arguments' ] = 'error';
+
+/**
+* Disallows unnecessary type assertions that do not change the type of an expression.
+*
+* @name no-unnecessary-type-assertion
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-assertion]{@link https://typescript-eslint.io/rules/no-unnecessary-type-assertion}
+*
+* @example
+* // Bad...
+* const x: number = 10;
+* const y = x as number;
+*
+* // Good...
+* const x: number = 10;
+* const y = x;
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-assertion' ] = 'error';
+
+/**
+* Disallows unused expressions which have no effect on the state of the program.
+*
+* @name no-unused-expressions
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unused-expressions]{@link https://typescript-eslint.io/rules/no-unused-expressions}
+*
+* @example
+* // Bad...
+* 0;
+* if(false) {}
+*
+* // Good...
+* (() => {
+*   const a = 0;
+* })();
+*/
+rules[ '@typescript-eslint/no-unused-expressions' ] = 'error';
+
+/**
+* Disables the rule against unused variables.
```

Comment:
Is this mainly to allow unused variables in function signatures? e.g., `foo( unused1, unused2, x, y )`?

Reply:
Hmm, I think we can turn that rule on. For function parameters, can use `argsignorepattern` [option](https://eslint.org/docs/latest/rules/no-unused-vars#argsignorepattern) and ignore parameters with leading `_`?

Reply:
That sounds reasonable. We don't typically name parameters with leading underscores, so probably don't need to use the `argsignorepattern` option.

Instead, we should use the `args` option and set to `after-used`.

Reply:
Done.

Reply:
Thanks!